v4/export/dump.go

```diff
@@ -206,3 +237,59 @@ func dumpTable(ctx context.Context, db *sql.DB, database, table string) (TableDa
 		colTypes: colTypes,
 	}, nil
 }
+
+func buildSelectAllQuery(conf *Config, db *sql.DB, database, table string) (string, error) {
+	var query strings.Builder
+	query.WriteString("SELECT * FROM ")
+	query.WriteString(database)
+	query.WriteString(".")
+	query.WriteString(table)
+	if conf.SortByPk {
+		orderByClause, err := buildOrderByClause(conf, db, database, table)
+		if err != nil {
+			return "", err
+		}
+		if orderByClause != "" {
+			query.WriteString(" ")
+			query.WriteString(orderByClause)
+		}
+	}
+	return query.String(), nil
+}
+
+func buildOrderByClause(conf *Config, db *sql.DB, database, table string) (string, error) {
+	pkName, err := getPrimaryKeyName(db, database, table)
+	if err != nil {
+		return "", err
+	}
+	if pkName != "" {
+		// the table has primary key.
+		return fmt.Sprintf("ORDER BY %s", pkName), nil
+	}
+	switch conf.ServerInfo.ServerType {
+	case TiDBServerType:
+		return "ORDER BY _tidb_rowid", nil
+	default:
+		// ignore when there is no primary key
+		return "", nil
+	}
+}
+
+const PrimaryKeyQuery = `SELECT COLUMN_NAME
+	FROM INFORMATION_SCHEMA.COLUMNS
+	WHERE TABLE_SCHEMA = '%s'
+	AND TABLE_NAME = '%s'
+	AND COLUMN_KEY = 'PRI';`
```

Comment:
Please use prepared statement for this, instead of `Sprintf`'ing the names. The table names can contain a `'`.

```sql
SELECT column_name FROM information_schema.columns
WHERE table_schema = ? AND table_name = ? AND column_key = 'PRI';
```